crates/router/src/core/payment_methods.rs

```
@@ -2650,3 +2667,169 @@ impl pm_types::SavedPMLPaymentsInfo {
+            recurring_mandate_payment_data: None,
+            preprocessing_id: None,
+            payment_method_balance: None,
+            connector_api_version: None,
+            request: payment_method_data_request.clone(),
+            response: Err(hyperswitch_domain_models::router_data::ErrorResponse::default()),
+            connector_request_reference_id: payment_method_session.id.get_string_repr().to_string(),
+            #[cfg(feature = "payouts")]
+            payout_method_data: None,
+            #[cfg(feature = "payouts")]
+            quote_id: None,
+            test_mode: None,
+            connector_http_status_code: None,
+            external_latency: None,
+            apple_pay_flow: None,
+            frm_metadata: None,
+            dispute_id: None,
+            refund_id: None,
+            connector_response: None,
+            payment_method_status: None,
+            minor_amount_captured: None,
+            integrity_check: Ok(()),
+            additional_merchant_data: None,
+            header_payload: None,
+            connector_mandate_request_reference_id: None,
+            authentication_id: None,
+            psd2_sca_exemption_type: None,
+        };
+
+    let payment_method_token_response = tokenization::add_token_for_payment_method(
+        &mut router_data,
+        payment_method_data_request.clone(),
+        state.clone(),
+        connector_id.clone(),
+        &merchant_connector_account_details.clone()
+    ).await?;
+
+    let token = payment_method_token_response
+                                    .payment_method_token_result.unwrap_or_else(|err| {
+                                    logger::error!("Failed to create token for payment method session confirm call : {:?}", err);
+                                    None
+                                });
+
+    logger::info!("Tokenization response");
+    logger::info!(token =? &token);
+
+    if let Some(token_str) = token {
```

if token is None, why we are silently discarding?

---

Not discarding now